### PR TITLE
fix(s13a.13): self-ref look-back no-prior short-circuit — Phase 6 #3f

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

Fix S13a.13: `a = ${?a}foo` with no prior `a` now resolves to `"foo"` (per HOCON.md L841), was `"foofoo"`.

Closes #84.

## Mechanism

When the resolver encounters a self-referential substitution `${?a}` (or `${a}`) inside the value being assigned to path `a`:
1. If a prior assignment to `a` exists → look-back returns the prior value (unchanged behavior).
2. **NEW**: If no prior exists → short-circuit. Optional `${?a}` → `undefined` (concat-fold omits it); required `${a}` → `ResolveError`.

Identification: a substitution is "self-referential in the value being constructed" when the **ConcatPlaceholder containing it is currently being iterated** by `resolveConcat`. Tracked via a `WeakSet<ConcatPlaceholder>` on the resolver.

## Why not the spec's path-equality?

The original spec ★1 decision #1 specified path-equality preservation. Round-2 multi-agent-review surfaced a false-positive: external lookups like `a = ${?a}foo; b = ${a}` would error because `b`'s `${a}` lookup found `a`'s concat-with-self-ref and triggered the short-circuit (wrongly).

All 3 impls (ts/rs/go) independently tightened the mechanism. ts uses node-membership-in-iterating-set (this PR); rs uses a path stack + `is_owner` guard; go uses AST-node pointer-identity. **All three are strictly narrower than path-equality** (correctly exclude external lookups).

Spec amendment deferred to a follow-up xx.hocon PR. See [xx.hocon#27](https://github.com/o3co/xx.hocon/issues/27) for the cross-impl tracking issue.

## Test plan

- [x] Conformance fixtures sr01–sr11 (xx.hocon `testdata/hocon/self-ref-lookback/`) all pass
- [x] Resolver-layer unit tests for 10 edge cases (no-prior, with-prior, required, array, nested, mutual-ref)
- [x] Round-2 regression test: `a = ${?a}foo; b = ${a}` → both `"foo"`
- [x] Existing pin test (`it.fails`) at `tests/resolver.test.ts:766` flipped to `it(...)` and passes
- [x] env-var-list/ev08 tripwire comments removed (cluster 3f has landed)
- [x] Full suite green: 734 pass, 21 expected-fail (unrelated spec rows), 2 skipped
- [x] No public API changes

## Pre-existing bugs surfaced (out of scope, see xx.hocon#27)

Round 2 review discovered 4 pre-existing resolver bugs beyond S13a.13 strict:
- Nested external ref crash (`foo.a = ${?foo.a}bar; foo.b = ${foo.a}`)
- Cache pollution on prior-with-external-ref (`a = "x"; a = ${?a}foo; b = ${a}`)
- Same-field double-self-ref crash (`a = ${?a}1; a = ${?a}2`)
- Order-dependent ref-before-self-ref (`b = ${a}; a = ${?a}foo`)

These are filed as cross-impl follow-ups; not addressed in this PR per the project's "unrelated issues → separate PR" rule.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/s13a13-self-ref-lookback.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  for (const name of SUCCESS_FIXTURES) {
+    it(`${name}: parses and resolves to expected JSON`, () => {
+      const content = readFileSync(join(confDir, `${name}.conf`), 'utf-8')
+      const expected = JSON.parse(
+        readFileSync(join(expectedDir, `${name}-expected.json`), 'utf-8'),
+      )
+      const config = parse(content)
+      expect(config.toObject()).toEqual(expected)
+    })
+  }
```
[the remark this answers is not in the document]
Comment (y1o1) [reply tests/s13a13-self-ref-lookback.test.ts:68]: Fixed in 6947aa2 — added existsSync skip-guard matching the pattern in tests/concat-errors.test.ts, tests/include-reservation.test.ts, and tests/env-var-list.test.ts. On fresh checkout the suite now skips gracefully instead of throwing ENOENT.

File: src/internal/resolver/substitution-resolver.ts [hunk 2]
```diff
@@ -149,36 +168,46 @@ export class SubstitutionResolver {
...
+              this.root,
+              s.segments.slice(0, s.segments.length - 1),
+            )
+            prior = parentScope?.priorValues.get(leafSeg)
+          } else {
+            const rootSeg = s.segments[0]?.text ?? ''
+            prior =
+              scope.priorValues.get(rootSeg) ??
+              this.root.priorValues.get(rootSeg)
+          }
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/internal/resolver/substitution-resolver.ts:200]: Fixed in 6947aa2 — aligned the cycle-detection branch's prior lookup criterion with the new short-circuit branch (`s.segments.length > 1`, not `s.prefixLen > 0`). Both branches now consistently handle dotted-path-at-root cases. Convergent with Claude r2 review I2 finding.
